core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -265,6 +265,32 @@ class PlaintextConsumerTest extends BaseConsumerTest {
     ensureNoRebalance(consumer, listener)
   }
 
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
+  @MethodSource(Array("getTestQuorumAndGroupProtocolParametersAll"))
+  def testMaxPollIntervalMsShorterThanPollTimeout(quorum: String, groupProtocol: String): Unit = {
+    this.consumerConfig.setProperty(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, 1000.toString)
+    this.consumerConfig.setProperty(ConsumerConfig.HEARTBEAT_INTERVAL_MS_CONFIG, 500.toString)
+
+    val consumer = createConsumer()
+    val listener = new TestConsumerReassignmentListener
+    consumer.subscribe(List(topic).asJava, listener)
+
+    // rebalance to get the initial assignment
+    awaitRebalance(consumer, listener)
+
+    val initialAssignedCalls = listener.callsToAssigned
+
+    consumer.poll(Duration.ofMillis(2000))
+
+    // Give enough time to rejoin
+    consumer.poll(Duration.ofMillis(500))
+    consumer.poll(Duration.ofMillis(500))
+
+    // Check that we did not rejoin
```

Comment:
Do we need this comment? I think it would be better to delete it and to rename `initialAssignedCalls` to something more meaningful like `callsToAssignedAfterFirstRebalance` or `callsToAssignedBeforePolls`.

Reply:
Done